### ISSUE TITLE
batch_processing fails on neurodebian

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#870]
3.0_beta10

~~~
brain@neurodebian:~/sct_example_data/mt$ sct_segment_graymatter -i mt0mt1.nii.gz -s mt1_seg.nii.gz

Spinal Cord Toolbox (version 3.0_beta10-bc4a1deefc59bbe3e58fa8a565f48763253c8d9d)
Running /home/brain/sct_3.0_beta10/scripts/sct_segment_graymatter.py -i mt0mt1.nii.gz -s mt1_seg.nii.gz

Building the appearance model...

Loading a reduced common space (using a PCA) ...

 ---> sum of eigenvalues : 25601230.092
kept eigenvalues (PCA space dimension)  : 47


-------> IN PCA : 

-> kept_modes:[[  2.39757031e-16  -8.21900381e-17   2.93683807e-17 ...,   4.24317548e-17
   -3.89564413e-17  -9.12340398e-17]
 [ -3.79002146e-16   3.60424298e-16  -3.35096744e-16 ...,  -2.00513278e-16
   -6.37809817e-16  -7.37722667e-16]
 [  8.81995193e-17  -1.86772023e-17  -4.35252228e-17 ...,  -1.35261546e-16
   -5.47676640e-16  -6.75053168e-16]
 ..., 
 [  2.74755125e-41  -3.70707367e-41  -1.68272897e-41 ...,   4.54141770e-42
   -7.48202727e-41   4.22570827e-41]
 [  1.75581766e-41  -3.29436870e-41   2.66076533e-41 ...,  -5.07662462e-41
    2.05743668e-40   1.53877001e-40]
 [ -1.10186724e-44   3.10061910e-42  -4.61501418e-43 ...,   9.75283339e-42
    3.05073686e-42   8.30540278e-42]]

-> kept_eigenval:47
mkdir tmp_mt0mt1_160514064027_50495/

Doing target pre-processing ...
cp mt0mt1.nii.gz tmp_mt0mt1_160514064027_50495//target.nii.gz
cp mt1_seg.nii.gz tmp_mt0mt1_160514064027_50495//target_sc_seg.nii.gz
sct_resample -i target.nii.gz -mm 0.3x0.3x17.0 -o target_resampled.nii.gz -x spline
sct_resample -i target_sc_seg.nii.gz -mm 0.3x0.3x17.0 -o target_sc_seg_resampled.nii.gz -x nn
sct_maths -i target_sc_seg_resampled.nii.gz -thr 0.0 -o target_sc_seg_resampled.nii.gz
sct_maths -i target_sc_seg_resampled.nii.gz -bin -o target_sc_seg_resampled.nii.gz
WARNING: File target_resampled.nii.gz already exists. Deleting it.
sct_image -i target_resampled.nii.gz -pad 39,39,0 -o target_resampled_pad.nii.gz
sct_image -i target_sc_seg_resampled.nii.gz -pad 39,39,0 -o target_sc_seg_resampled_pad.nii.gz
sct_image -i target_resampled_pad.nii.gz -setorient RPI -o target_resampled_pad_RPI.nii.gz
sct_image -i target_sc_seg_resampled_pad.nii.gz -setorient RPI -o target_sc_seg_resampled_pad_RPI.nii.gz
sct_crop_image -i target_resampled_pad_RPI.nii.gz -m target_sc_seg_resampled_pad_RPI.nii.gz -b 0 -o target_resampled_pad_RPI_seg_in.nii.gz
sct_create_mask -i target_resampled_pad_RPI_seg_in.nii.gz -p centerline,target_sc_seg_resampled_pad_RPI.nii.gz -size 75 -o target_resampled_pad_RPI_square_mask_from_sc_seg.nii.gz -f box
WARNING: an error occured when croping target_resampled_pad_RPI... 
 could not broadcast input array from shape (618,618) into shape (617,617)
Traceback (most recent call last):
  File "/home/brain/sct_3.0_beta10/scripts/sct_segment_graymatter.py", line 694, in <module>
    gmsegfull.segment()
  File "/home/brain/sct_3.0_beta10/scripts/sct_segment_graymatter.py", line 354, in segment
    self.segmentation_pipeline()
  File "/home/brain/sct_3.0_beta10/scripts/sct_segment_graymatter.py", line 391, in segmentation_pipeline
    self.preprocessed.process()
  File "/home/brain/sct_3.0_beta10/scripts/sct_segment_graymatter.py", line 310, in process
    self.square_mask, self.processed_target = crop_t2_star(self.t2star, self.sc_seg, box_size=box_size)
  File "/home/brain/sct_3.0_beta10/scripts/msct_gmseg_utils.py", line 691, in crop_t2_star
    return mask_box, fname_seg_in_IRP
UnboundLocalError: local variable 'fname_seg_in_IRP' referenced before assignment
~~~

**Comment by jcohenadad** [comment]
test passes:

~~~
Spinal Cord Toolbox (version 3.0_beta10-bc4a1deefc59bbe3e58fa8a565f48763253c8d9d)
Running /home/brain/sct_3.0_beta10/scripts/sct_testing.py -f sct_segment_graymatter

Downloading testing data...
sct_download_data -d sct_testing_data

Path to testing data: /home/brain/sct_example_data/mt/sct_testing_data/data/
Checking test_sct_segment_graymatter................[OK]
status: [0]
Finished! Elapsed time: 21s


Remove temporary files...
rm -rf tmp.160514064338/
~~~

so this bug is data-dependent

**Comment by jcohenadad** [comment]
@SaraDupont: is it the same bug that was observed at the SCT course?

**Comment by SaraDupont** [comment]
Yes, it is exactly the same error. 
During the SCT course, the error was observed with version 2.06. The user had a VM (Oracle v5.0.20) and his laptop was a Mac running on Yosemite 10.10.5.
Data are on the USB stick SCT-1 : we should check if this is data-dependent (reproducible on the same data on another system) or system-dependent. It could also be due to an error during the processings before sct_segment_graymatter that produces corrupted data : to be checked also.

**Comment by poquirion** [comment]
fname_seg_in_IRP has to be set before the `try` block

**Comment by SaraDupont** [comment]
Blocked by issue #898 (Can't test on NeuroDebian)

**Comment by SaraDupont** [comment]
I reproduced the error on OSX: I tried to segment a T2* in the template space using the cord segmentation from the template (ie. MNI-Poly-AMU_cord.nii.gz) which is not "full" in the superior-inferior direction (Z).
The issue was related to the creation of a square mask to crop data in the preprocessing ==> the mask created didn't have the same size as the data along Z, which created an issue when using the mask to crop and stack slices. 
This should be fixed by fixing #879

**Comment by SaraDupont** [comment]
Possible fix with commits 2c192fc and b43b1da: error is fixed on OSX but I don't know about linux/VM (I have installation issues on my NeuroDebain and can't test anything: see issue  #898). 
@jcohenadad Could you re-test on NeuroDebian if the error persists ?

**Comment by SaraDupont** [comment]
I re-tested this with new version of `sct_segment_graymatter` and it worked fine.
